src/clojure/nrepl/cmdline.clj

```diff
@@ -239,64 +249,190 @@
-            handler (some->> (:handler options) (require-and-resolve :handler))
-            handler (or handler (build-handler middleware))
-            greeting-fn (if (= transport #'transport/tty) #'transport/tty-greeting)
-            server (start-server :port port :bind bind :handler handler
-                                 :transport-fn transport :greeting-fn greeting-fn)]
-        (when-let [ack-port (some-> (:ack options) ->int)]
-          (binding [*out* *err*]
-            (println (format "ack'ing my port %d to other server running on port %d"
-                             (:port server) ack-port)
-                     (:status (send-ack (:port server) ack-port transport)))))
-        (let [port (:port server)
-              ^java.net.ServerSocket ssocket (:server-socket server)
-              host (.getHostName (.getInetAddress ssocket))]
-          ;; The format here is important, as some tools (e.g. CIDER) parse the string
-          ;; to extract from it the host and the port to connect to
-          (println (format "nREPL server started on port %d on host %s - %s://%s:%d"
-                           port host (transport/uri-scheme transport) host port))
-          ;; Many clients look for this file to infer the port to connect to
-          (let [port-file (io/file ".nrepl-port")]
-            (.deleteOnExit port-file)
-            (spit port-file port))
-          (if (:interactive options)
-            (run-repl host port (merge (when (:color options) colored-output)
-                                       {:transport transport}))
-            ;; need to hold process open with a non-daemon thread -- this should end up being super-temporary
-            (Thread/sleep Long/MAX_VALUE)))))))
+        merge-config (partial merge config/config)
+        options (-> options
+                    (keywordize-options)
+                    (parse-cli-values)
+                    (merge-config))]
+    [options _args]))
+
+(defn display-help
+  "Displays help copy to the user and exits the program"
+  []
+  (println (help))
+  (exit 0))
+
+(defn display-version
+  "Displays the version to the user and exits the program"
+  []
+  (println (:version-string version/version))
+  (exit 0))
+
+(defn- resolve-transport
+  "Takes a map of nREPL CLI options.
+  Returns either a default transport or the
+  value of :transport."
+  [options]
+  (or (some->> options
+               (:transport)
+               (require-and-resolve :transport))
+      #'transport/bencode))
+
+(defn- resolve-handler
+  "Takes a map of nREPL CLI options and list of middleware.
+  Returns a request handler function.
+  If some handler was explicitly passed we'll use it, otherwise we'll build
+  one from whatever was passed via --middleware"
+  [options middleware]
+  (or (some->> options
+               (:handler)
+               (require-and-resolve :handler))
+      (build-handler middleware)))
+
+(defn- resolve-ack-port
```

Comment:
Only functions that do var resolution should have `resolve` in their names.

Reply:
What would you suggest as name for such a function? Previously it was `get-ack-port-option` but that may be too Java-y with the extra verbiage. 

How about `options->ack-port`?